src/components/split-axis-events.js

```diff
@@ -1,30 +1,376 @@
+    }
+
+    this.actuator.pulse(strength, duration);
+  }
+});
+
+AFRAME.registerComponent("vive-controls-extended", {
+  schema: {
+    hand: { default: "left" },
+    dpad: { default: false },
+    dpad_livezone: { default: 0.3 },
+    dpad_deadzone: { default: 0.8 },
+    dpad_directions: { default: 4 },
+    dpad_turbo: { default: true },
+    dpad_pressed_turbo: { default: false },
+    center_zone: { default: 0.3 },
+    dpad_haptic_intensity: { default: "none" },
+    dpad_pressed_haptic_intensity: { default: "none" }
+  },
+
   init: function() {
-    this.pressed = false;
+    this.dpadCanFire = true;
+    this.dpadPressedCanFire = true;
+    this.trackpadPressed = false;
     this.onAxisMove = this.onAxisMove.bind(this);
+    this.lastSeenAxes = [0, 0];
     this.onButtonChanged = this.onButtonChanged.bind(this);
   },
 
   play: function() {
     this.el.addEventListener("axismove", this.onAxisMove);
-    this.el.addEventListener("buttonchanged", this.onButtonChanged);
+    this.el.addEventListener("trackpadchanged", this.onButtonChanged);
   },
 
   pause: function() {
     this.el.removeEventListener("axismove", this.onAxisMove);
-    this.el.removeEventListener("buttonchanged", this.onButtonChanged);
+    this.el.removeEventListener("trackpadchanged", this.onButtonChanged);
   },
 
   onAxisMove: function(event) {
-    var name = "touchpad" + (this.pressed ? "pressed" : "") + "axismove";
-    this.el.emit(name + "x", event.detail.axis[0]);
-    this.el.emit(name + "y", event.detail.axis[1]);
+    var x = event.detail.axis[0];
+    var y = event.detail.axis[1];
+    this.lastSeenAxes = [x, y];
+    var hand = this.data.hand;
+    const axisMoveEvent = `${hand}_trackpad${this.trackpadPressed
+      ? "_pressed"
+      : ""}_axismove`;
+    this.el.emit(axisMoveEvent, {
+      axis: [x, y]
+    });
+
+    var deadzone = this.data.dpad_deadzone;
+    var turbo = this.data.dpad_turbo;
+    var pressedTurbo = this.data.dpad_pressed_turbo;
+    var livezone = this.data.dpad_livezone;
+    var directions = this.data.dpad_directions;
+    var hapticIntensity = this.data.dpad_haptic_intensity;
+    var pressedHapticIntensity = this.data.dpad_pressed_haptic_intensity;
+
+    if (!turbo && Math.abs(x) < livezone && Math.abs(y) < livezone) {
+      this.dpadCanFire = true;
+    }
+
+    event.target.emit(`${hand}_haptic_pulse`, {
+      intensity: this.trackpadPressed ? pressedHapticIntensity : hapticIntensity
+    });
+
+    x = Math.abs(x) < deadzone ? 0 : x;
+    y = Math.abs(y) < deadzone ? 0 : y;
+    if (x == 0 && y == 0) return;
+    var angle = Math.atan2(x, y);
+    var direction =
+      directions === 4 ? angleTo4Direction(angle) : angleTo8Direction(angle);
+
+    if (!this.trackpadPressed && !this.dpadCanFire) {
+      return;
+    }
+    if (this.trackpadPressed && !this.dpadPressedCanFire) {
+      return;
+    }
+    var dpadEvent = `${hand}_trackpad_dpad${this.trackpadPressed
+      ? "_pressed"
+      : ""}_${direction}`;
```

Comment:
Extracting the ternary expression into a separate variable would be more readable.